x/auth/tx/direct_aux.go

```diff
@@ -0,0 +1,62 @@
+package tx
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/types/tx"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// signModeDirectAuxHandler defines the SIGN_MODE_DIRECT_AUX SignModeHandler
+type signModeDirectAuxHandler struct{}
+
+var _ signing.SignModeHandler = signModeDirectAuxHandler{}
+
+// DefaultMode implements SignModeHandler.DefaultMode
+func (signModeDirectAuxHandler) DefaultMode() signingtypes.SignMode {
+	return signingtypes.SignMode_SIGN_MODE_DIRECT_AUX
+}
+
+// Modes implements SignModeHandler.Modes
+func (signModeDirectAuxHandler) Modes() []signingtypes.SignMode {
+	return []signingtypes.SignMode{signingtypes.SignMode_SIGN_MODE_DIRECT_AUX}
+}
+
+// GetSignBytes implements SignModeHandler.GetSignBytes
+func (signModeDirectAuxHandler) GetSignBytes(mode signingtypes.SignMode, data signing.SignerData, tx sdk.Tx) ([]byte, error) {
+	if mode != signingtypes.SignMode_SIGN_MODE_DIRECT_AUX {
+		return nil, fmt.Errorf("expected %s, got %s", signingtypes.SignMode_SIGN_MODE_DIRECT_AUX, mode)
+	}
+
+	protoTx, ok := tx.(*wrapper)
+	if !ok {
+		return nil, fmt.Errorf("can only handle a protobuf Tx, got %T", tx)
+	}
+
+	signDocDirectAux := types.SignDocDirectAux{
+		BodyBytes:     protoTx.getBodyBytes(),
+		ChainId:       data.ChainID,
+		AccountNumber: data.AccountNumber,
+		Sequence:      data.Sequence,
+		Tip:           protoTx.tx.AuthInfo.Tip,
+		// PublicKey: ,
```

Comment:
Ah I see, we don't know the public key of the signer here.

I propose to:
```diff
type SignerData struct {
	ChainID string
	AccountNumber uint64
	Sequence uint64
+	PubKey crypotypes.PubKey // add a comment that's it's only useful for SIGN_MODE_AUX
}
```

Reply:
cc @aaronc Do you have any other ideas?

Reply:
Actually if we go with `SignerIndex`, is there a risk that users forget to populate this field in `SignerData`, and that it get sets to `0`, which can lead to bugs?

In which case `PubKey` might be more explicit and less bug-prone?